### ISSUE TITLE
Firefox extensions

### ISSUE THREAD

**Issue by zalmoxisus** [zalmoxisus/browser-redux#12]
For building Firefox extensions we use [WebExtensions](https://wiki.mozilla.org/WebExtensions)'s standards, which is still supported only in Firefox Nightly.

To completely support it, we need:
[ ] a proof of concept
[ ] tests with Selenium using Firefox Nightly
[ ] support in Firefox release as planned to be on [December 15th (Firefox 43 merges to release)](https://blog.mozilla.org/addons/2015/08/21/the-future-of-developing-firefox-add-ons/)
[ ] [support publishing WebExtensions to addons.mozilla.org](https://bugzilla.mozilla.org/show_bug.cgi?id=1210037).

**Comment by jhen0409** [comment]
https://support.mozilla.org/en-US/kb/add-on-signing-in-firefox

The Firefox cannot use unsigned extension now, we should recommend use [Firefox Developer Edition](https://developer.mozilla.org/en-US/Firefox/Developer_Edition) and disable `xpinstall.signatures.required`. 

Also, the Firefox Developer Edition which can load local directories.

**Comment by zalmoxisus** [comment]
@jhen0409, Yes, AFAIK, Firefox doesn't support webextensions at all yet, and we need whether Nightly or Developer Edition. About signing it is indicated in [the load section](https://github.com/zalmoxisus/crossbuilder#load). I know the readme needs rework :) The original plan of Mozilla was to migrate to WebExtensions by December last year, but it is taking longer, and for now it is difficult to implement something as the API is too limited.

**Comment by groovecoder** [comment]
Just FYI here, Firefox has implemented a large number of web extensions API since February ...

http://arewewebextensionsyet.com/

**Comment by zalmoxisus** [comment]
@groovecoder, thanks for the info. Yes, the API is improving. We did support firefox webextension's api in ['v0.3.1'](https://github.com/zalmoxisus/crossbuilder/releases/tag/v0.3.1). Specifically for the current boilerplate we also need `chrome.runtime.getBackgroundPage` to be supported. Strange that according to that page it is already supported, but according the [the issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1213473) it's not. I'll look into it.

**Comment by groovecoder** [comment]
I used it recently so I'm fairly sure it's supported now. That bug may still be open because of some other `chrome.runetime` method/property that isn't done yet? 😕
